### ISSUE TITLE
Azure storage uploading not working with SDK 1.9.0

### ISSUE THREAD

**Issue by cajewsa** [allegroai/clearml#868]
## Describe the bug
Azure storage upload does not work with 1.9.0, even though it works with 1.8.3.

## To reproduce
Using `AzureContainerConfig` to set up output_uri:
```python
def get_storage_uri() -> str:
    # Set up logging to Azure
    bucket_config = AzureContainerConfig(
        account_name="💩",  # type: ignore
        account_key="💩",  # type: ignore
        container_name="💩",  # type: ignore
    )
    StorageHelper.add_azure_configuration(bucket_config)
    return StorageHelper.get_azure_storage_uri_from_config(bucket_config)


def init_experiment_tracker(config: DictConfig) -> Task:
    """Initialize the experiment tracker and return the task and
    (possibly modified) configuration object."""

    # Initialize the task
    storage_uri = get_storage_uri()
    task: Task = Task.init(
        project_name=config.clearml.project_name,
        task_name=config.clearml.task_name,
        task_type=Task.TaskTypes.training,
        reuse_last_task_id=False,
        output_uri=storage_uri,
    )
    task.logger.set_default_upload_destination(storage_uri)
    return task
```

## Expected behaviour
This should upload all output and artifacts to Azure, instead of default file server.
It works flawlessly on my setup with `clearml==1.8.*` but when upgrading to `clearml==1.9.*` it fails with the error:
```
Traceback (most recent call last):
  File "/home/azureuser/.local/share/virtualenvs/daily-representation-F_rNT0sY/lib/python3.10/site-packages/clearml/storage/helper.py", line 999, in check_write_permissions
    self.delete(path=dest_path)
  File "/home/azureuser/.local/share/virtualenvs/daily-representation-F_rNT0sY/lib/python3.10/site-packages/clearml/storage/helper.py", line 984, in delete
    return self._driver.delete_object(self.get_object(path))
  File "/home/azureuser/.local/share/virtualenvs/daily-representation-F_rNT0sY/lib/python3.10/site-packages/clearml/storage/helper.py", line 2230, in delete_object
    container = object.container
AttributeError: 'NoneType' object has no attribute 'container'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/azureuser/.local/share/virtualenvs/daily-representation-F_rNT0sY/lib/python3.10/site-packages/clearml/binding/hydra_bind.py", line 173, in _patched_task_function
    return task_function(a_config, *a_args, **a_kwargs)
  File "/home/azureuser/daily-representation/src/train_vae.py", line 20, in main
    task = init_experiment_tracker(cfg)
  File "/home/azureuser/daily-representation/src/core/utils/hydra.py", line 48, in init_experiment_tracker
    task: Task = Task.init(
  File "/home/azureuser/.local/share/virtualenvs/daily-representation-F_rNT0sY/lib/python3.10/site-packages/clearml/task.py", line 620, in init
    task.output_uri = output_uri
  File "/home/azureuser/.local/share/virtualenvs/daily-representation-F_rNT0sY/lib/python3.10/site-packages/clearml/task.py", line 1135, in output_uri
    helper.check_write_permissions(value)
  File "/home/azureuser/.local/share/virtualenvs/daily-representation-F_rNT0sY/lib/python3.10/site-packages/clearml/storage/helper.py", line 1001, in check_write_permissions
    raise ValueError('Insufficient permissions (delete failed) for {}'.format(base_url))
ValueError: Insufficient permissions (delete failed) for azure://💩.blob.core.windows.net/💩
```

## Environment
* Server type: self hosted
* ClearML SDK: 1.8.3 vs. 1.9.0
* ClearML Server Version: 1.7.0-232
* Python Version: 3.10.6
* OS: Linux
## Related Discussion
Likely related to https://github.com/allegroai/clearml/issues/647
Obviously 💩 are not the real configuration values :-)

**Comment by jkhenning** [comment]
Hi @cajewsa,

We're taking a look, will update here once we have some results 🙂

**Comment by pollfly** [comment]
Hey @cajewsa! Just letting you know that this issue has been resolved in the recently released [v1.9.1](https://github.com/allegroai/clearml/releases/tag/v1.9.1). Let us know if there are any issues :)
